python_modules/dagster/dagster/_core/execution/context/output.py

```diff
@@ -208,13 +219,20 @@ def run_id(self) -> str:
 
         return self._run_id
 
+    @deprecated(breaking_version="2.0.0", additional_warn_text="Use definition_metadata instead")
     @public
     @property
     def metadata(self) -> Optional[ArbitraryMetadataMapping]:
+        """Deprecated: used definition_metadata instead."""
+        return self._definition_metadata
+
+    @public
+    @property
+    def definition_metadata(self) -> ArbitraryMetadataMapping:
```

Comment:
I took the opportunity to tighten up the typing for this property. `definition_metadata` will never be None (we always set it to an empty dictionary at minimum)